src/main/java/uk/gov/gchq/magmacore/database/MagmaCoreObjectDatabase.java

```diff
@@ -135,6 +136,47 @@ public void dump(final PrintStream out) {
      * @param out Output stream to dump data to.
      */
     public final void dumpTurtle(final PrintStream out) {
-        objects.values().forEach(object -> out.println(object.toTriples()));
+        objects.values().forEach(object -> out.println(Triples.toTriples(object)));
+    }
+
+    /**
+     * Begin a writeable transaction initially in READ mode,
+     * but in Jena it will switch to WRITE mode if updates are made.
+     *
+     * @return the {@link MagmaCoreDatabase}
+    */
+    public final MagmaCoreDatabase begin() {
```

Comment:
If client code is going to allow it's database to be any of the supported implementations (Remote, Local, Object) then those databases are going to have to have the same interface otherwise the client will have to know what kind of database it is using so it can decide whether or not to use these methods. It is possible to provide default implementations in the `MagmaCoreDatabase` interface, but not doing so forces the implementer to decide how to handle these methods rather than letting them accept the default. I can add defaults to the interface and remove these if you really think it's necessary.

Reply:
I prefer forcing the implementing class to handle these methods, even if involves explicitly disabling this function like this. I just wish there was a way to conceal these unused methods, but that's not how Java works 😞